src/options/utils/dragging.js

```diff
@@ -0,0 +1,194 @@
+let scrollEdgeBottom;
+let scrollTimer;
+let scrollSpeed;
+let scrollTimestamp;
+let xyCache;
+
+/**
+ * @param {Element} el
+ * @param {function(from,to)} onDrop
+ */
+export default function enableDragging(el, { onDrop }) {
+  if (!parent) {
+    parent = el.parentElement;
+    // pre-FF64 doesn't support `@media (pointer: coarse)`
+    if (isTouch && !matchMedia('(pointer: coarse)').matches) {
+      parent.classList.add('touch');
+    }
+  }
+  el::on(eventNames.start, isTouch ? onTouchStart : onDragStart);
+  el::on(DROP_EVENT_RELAY, () => onDrop(index, lastIndex));
+}
+
+function onTouchStart(e) {
+  original = this;
+  longPressEvent = e;
+  longPressTimer = setTimeout(onTouchMoveDetect, LONGPRESS_DELAY, 'timer');
+  document::on(eventNames.move, onTouchMoveDetect);
+  document::on(eventNames.end, onTouchEndDetect);
+}
+
+function onTouchMoveDetect(e) {
+  onTouchEndDetect();
+  if (e === 'timer') {
+    original::onDragStart(longPressEvent);
+    if (isFirefox && parentCanScroll()) {
+      // FF bug workaround: prevent the script list container from scrolling on drag
+      parent.scrollTop += 1;
+      parent.scrollTop -= 1;
+    }
+  }
+}
+
+function onTouchEndDetect() {
+  clearTimeout(longPressTimer);
+  document::off(eventNames.move, onTouchMoveDetect);
+  document::off(eventNames.end, onTouchEndDetect);
+}
+
+function onDragStart(e) {
+  original = this;
+  if (e.cancelable) e.preventDefault();
+  const { clientX: x, clientY: y } = e.touches?.[0] || e;
+  const rect = original.getBoundingClientRect();
+  const parentRect = parent.getBoundingClientRect();
+  dragged = original.cloneNode(true);
+  elements = [...parent.children];
+  index = elements.indexOf(original);
+  lastIndex = index;
+  elements.splice(index, 1);
+  height = rect.height;
+  offsetX = x - rect.left;
+  offsetY = y - rect.top;
+  scrollEdgeTop = parentRect.top + SCROLL_GAP;
+  scrollEdgeBottom = parentRect.bottom - SCROLL_GAP;
+  xyCache = {};
+  original.classList.add('dragging-placeholder');
+  dragged.classList.add('dragging');
+  dragged.style.transform = `translate(${rect.left}px, ${rect.top}px)`;
+  dragged.style.width = `${rect.width}px`;
+  parent.appendChild(dragged);
+  if (isTouch) parent.insertAdjacentElement('afterBegin', noScroll);
+  document::on(eventNames.move, onDragMouseMove);
+  document::on(eventNames.end, onDragMouseUp);
+}
+
+function onDragMouseMove(e) {
+  const { clientX: x, clientY: y, target } = e.touches?.[0] || e;
+  let moved;
+  const hovered = isTouch ? scriptFromPoint(x, y) : target.closest('.script');
+  // FF bug: despite placeholder having `pointer-events:none` it's still reported in `target`
+  if (hovered && hovered !== original) {
+    const rect = hovered.getBoundingClientRect();
+    const isDown = y > rect.top + rect.height / 2;
+    moved = original !== hovered[`${isDown ? 'next' : 'previous'}ElementSibling`];
+    if (moved) {
+      hovered.insertAdjacentElement(isDown ? 'afterEnd' : 'beforeBegin', original);
```

Comment:
The original element does not need to be moved during dragging.

Reply:
I thought I preserved the old internal behavior. Anyway it seems to work correctly in the sense that I couldn't break it :smile: